### ISSUE TITLE
Navigation: Creating new page from within the linker shall set title and slug right away (currently sets numerical ID)

### ISSUE THREAD

**Issue by porg** [WordPress/gutenberg#46731]
- Followup to #18900
  - @getdave: I ping you, as you implemented that feature and maybe also have interest in that followup.

In the [mockup video](https://github.com/WordPress/gutenberg/issues/18900#issuecomment-578200190) by @shaunandrews one sees as the feature should work:
1. In the linker you type in your word(s), e.g. "Our Story"
2. In the results below the Combobox, you click the option "New Page: Our Story"
3. The page get's created.
  - ✅ With the page/blogpost metadata title "Our Story".
  - ✅ With the h1 prefilled to "Our Story", you can choose to hide/remove/alter later if you want.
  -  With the proper slug `/our-story/`
    - ✅ In the mockup video this is the case.
    - ❌ In WP 6.1.1 this is not the case, instead it get's created as `/?page_id=1234`
      - You then have to open the page, and edit the slug there manually.
      -  The "ad-hoc link/page creation mode" is supposed to provide you a distraction-free working mode.
      - This unnecessary administrative burden destroys that again.
      - The slug should be created automatically.

**Comment by porg** [comment]
#29395 would be fixed too by creating the slug right away too.
- Does the Wordpress architecture allow to create the slug right away to reserve it in the namespace, but nevertheless not yet publish that page/blogpost automatically, but keeping it as a draft, as it is a skeleton only (title but nothing else)?
- Even if not, I think it would be the right way.
- Because if people are in the process of creating a website structure and not wanting to expose their website they can
  - run it locally or on a LAN,
  - keep it from web spiders of mainstream search engines which respect robots.txt with the built-in preference: Settings > Reading Settings > Search engine visibility > ☑︎ Discourage search engines from indexing this site
  - or keep the website login-only with a free plugin like [My Private Site](https://wordpress.org/plugins/jonradio-private-site/)

👉 Quite many good reasons to create and reserve the slug right away.

**Comment by getdave** [comment]
Thanks for your proposal. 
  
> * Does the Wordpress architecture allow to create the slug right away to reserve it in the namespace, but nevertheless not yet publish that page/blogpost automatically, but keeping it as a draft, as it is a skeleton only (title but nothing else)?

I wonder if @anton-vlasenko or @ironprogrammer would know the answer to this? 

I'll note that it's unlikely we'll be able to change the way WordPress Core works to accommodate a UX requirement of the Navigation block.

**Comment by anton-vlasenko** [comment]
> Does the Wordpress architecture allow to create the slug right away to reserve it in the namespace, but nevertheless not yet publish that page/blogpost automatically, but keeping it as a draft, as it is a skeleton only (title but nothing else)?
I wonder if @anton-vlasenko or @ironprogrammer would know the answer to this?

Yes, I agree with @getdave.
As far as I know, there is no "reliable" way to reserve a slug in WordPress.
When publishing a post, WordPress checks if other posts, including drafts, already use the same slug. If a conflict is found, WordPress appends a numeric suffix (e.g., `-2`, `-3`, etc.) to create a unique slug for the new post.
However, if the current draft has a slug not used by other posts, WordPress will use that slug when publishing the post.

In summary, slug names aren’t finalized until the post is published, which is when WordPress officially assigns the slug.
Inspecting the source code of the [wp_unique_post_slug()](https://developer.wordpress.org/reference/functions/wp_unique_post_slug/) function might provide more insight into the specific details of the slug assignment mechanism.

**Comment by ntsekouras** [comment]
Thanks for the report! I think this was handled by #71188, which added a dedicated Create page step to the Link UI with a Publish option that is on by default, so the page is published with its slug right away. Drafts still get the ?page_id= link because WordPress can't reserve a slug before publishing, as explained above. Feel free to reopen if you still see it on trunk.
